crate/cli/src/tests/kms/xml/kmip_2_1/compare.rs

```diff
@@ -0,0 +1,282 @@
+use cosmian_kmip::kmip_2_1::{self};
+use cosmian_kms_client::cosmian_kmip::kmip_2_1::kmip_operations::Operation;
+
+use crate::{
+    error::{KmsCliError, result::KmsCliResult},
+    tests::kms::xml::compare::{compare_attributes, compare_object},
+};
+
+pub(crate) fn compare_payload_v21(expected: &Operation, actual: &Operation) -> KmsCliResult<()> {
+    use std::mem::discriminant;
+    if discriminant(expected) != discriminant(actual) {
+        return Err(KmsCliError::Default(format!(
+            "Operation type mismatch expected={expected} actual={actual}"
+        )));
+    }
+
+    match (expected.clone(), actual.clone()) {
+        (Operation::GetResponse(exp), Operation::GetResponse(act)) => {
+            if exp.unique_identifier != act.unique_identifier {
+                // Some servers set GetResponse.unique_identifier to the wrapping key's UID
+                // when KeyWrappingData is present. Tolerate this specific divergence by
+                // checking if the actual UID equals the EncryptionKeyInformation UID.
+                use kmip_2_1::kmip_objects::Object as Obj;
+
+                fn wrapping_encryption_uid(
+                    obj: &kmip_2_1::kmip_objects::Object,
+                ) -> Option<kmip_2_1::kmip_types::UniqueIdentifier> {
+                    match obj {
+                        Obj::SymmetricKey(sk) => sk
```

Comment:
I guess this could be collapsed this way:

```
Obj::SymmetricKey(key) |  Obj::PrivateKey(key) | Obj::PublicKey(key) | ... => {
                           key
                            .key_block
                            .key_wrapping_data
                            .as_ref()
                            .and_then(|kwd| kwd.encryption_key_information.as_ref())
                            .map(|eki| eki.unique_identifier.clone()),
                            ```